diff -aur valgrind-3.11.0/Makefile.am valgrind-3.11.0/Makefile.am
--- valgrind-3.11.0/Makefile.am	2015-09-08 15:23:31.000000000 +0200
+++ valgrind-3.11.0/Makefile.am	2016-03-07 14:24:44.044735899 +0100
@@ -10,6 +10,7 @@
 		lackey \
 		none \
 		helgrind \
+        tracergrind \
 		drd
 
 EXP_TOOLS = 	exp-sgcheck \
diff -aur valgrind-3.11.0/auxprogs/gen-mdg valgrind-3.11.0/auxprogs/gen-mdg
--- valgrind-3.11.0/auxprogs/gen-mdg	2015-09-08 15:23:30.000000000 +0200
+++ valgrind-3.11.0/auxprogs/gen-mdg	2016-03-07 14:27:06.229597050 +0100
@@ -55,7 +55,7 @@
 
 # List of all tools.
 my @tools = ( "cachegrind", "helgrind",
-              "lackey", "massif", "memcheck", "none" );
+              "lackey", "massif", "memcheck", "none", "tracergrind" );
 
 my $usage = <<END
 usage: gen-mdg [options]
diff -aur valgrind-3.11.0/configure.ac valgrind-3.11.0/configure.ac
--- valgrind-3.11.0/configure.ac	2015-09-22 22:26:31.000000000 +0200
+++ valgrind-3.11.0/configure.ac	2016-03-07 14:25:34.268608902 +0100
@@ -4099,6 +4099,8 @@
    exp-dhat/tests/Makefile
    shared/Makefile
    solaris/Makefile
+   tracergrind/Makefile
+   tracergrind/tests/Makefile
 ])
 AC_CONFIG_FILES([coregrind/link_tool_exe_linux],
                 [chmod +x coregrind/link_tool_exe_linux])
